src/ifcopenshell-python/ifcopenshell/express/schema_class.py

```diff
@@ -311,6 +315,7 @@ def can_be_instantiated(idx_name):
         self.statements[self.statements.index("{factory_placeholder}")] = (
             """
 class %(schema_name)s_instance_factory : public IfcParse::instance_factory {
+    virtual ~%(schema_name)s_instance_factory() { s = 0; }
```

Comment:
I think I understand why you chose this approach, but it does seem a bit odd to have such side effects in a destructor. If we keep it, I think it deserves a quick comment on why and the implications.

Perhaps the line of control needs to be reversed. For example:
- Make `s` a `std::unique_ptr<schema_definition>`.
- Instead of having the destructor clear `s`, why not have a `clear_schema()` function in this namespace that does `s.reset(nullptr)`. 
- If `s` becomes global scope, maybe it also deserves a better name.